### PR TITLE
Use kpm list in makefile

### PR DESCRIPTION
**Pull request by troydai**

### CODE REVIEW

**Comment by troydai** [comment]
@davidfowl

File: makefile.shade [hunk 1]
```diff
@@ -628,85 +541,97 @@ var RC_FILES = '${FindAllFiles("Resource.rc", BOOTSTRAPPER_EXE_NAME, BOOTSTRAPPE
...
+                        if (runtimeAssemblies.Add(file))
+                        {
+                            var targetFilePath = Path.Combine(target.TargetFolder, Path.GetFileName(file));
+                            Log.Info("Copying " + file + " to " + targetFilePath);
+                            File.Copy(file, targetFilePath, true);
+                        }
                     }
                 }
-                else
+                catch (Exception ex)
```
Comment (davidfowl) [inline makefile.shade]: When would this throw?

File: makefile.shade [hunk 2]
```diff
@@ -751,6 +698,62 @@ functions @{
...
+        }
+
+        int result;
+        if (!IsMono)
+        {
+            result = ExecuteAndRedirectOutput("cmd", string.Format("/C kpm {0}", argument), tmpFile);
+        }
+        else 
+        {
+            result = ExecuteAndRedirectOutput("kpm", argument, tmpFile);
```
Comment (davidfowl) [inline makefile.shade]: Call it ``ExecuteAndCaptureOutput `. What about exit codes? Should it return a tuple?

File: makefile.shade [hunk 3]
```diff
@@ -628,85 +541,97 @@ var RC_FILES = '${FindAllFiles("Resource.rc", BOOTSTRAPPER_EXE_NAME, BOOTSTRAPPE
...
-            
-            var runtimeDeps = deps["Runtime"];
-            
-            foreach(var targetClr in RUNTIME_CORECLR_TARGETS)
+
+            // copy required CoreCLR assemblies
+            if (target.Framework == "aspnetcore50")
             {
-                foreach(var file in Directory.EnumerateFiles(targetClr, "System.*.dll"))
+                foreach (var file in Directory.GetFiles(coreclrFolder).Where(f => !Path.GetFileName(f).StartsWith("System")))
```
Comment (davidfowl) [inline makefile.shade]: Why this change?

**Comment by davidfowl** [comment]
:shipit:

File: makefile.shade (hunk 1 above)
Comment (troydai) [inline makefile.shade]: File copy failure.

File: makefile.shade (hunk 3 above)
Comment (troydai) [inline makefile.shade]: These are different. Originally it copies all files and then go delete `System.*.dll` if it doesn't exist in dependency list. Now it copy all files except `System.*.dll` which are copied in previous logic.

**Comment by troydai** [comment]
rebase and verified. will merge in seconds
